### ISSUE TITLE
TWIR 121's CotW + PRs

### ISSUE THREAD

**Issue by llogiq** [cmr/this-week-in-rust#176]
This week's Crate of the Week is [preferences](https://crates.io/crates/preferences) which does the right thing to your program's preferences on all common operating systems.

Interesting PRs:

https://github.com/rust-lang/rust/pull/32055 no more private imports in rustdoc
https://github.com/rust-lang/rust/pull/32039 rustc will catch missing fields in pattern matches again
https://github.com/rust-lang/rust/pull/32032 regression fix for LLVM type errors with bools
https://github.com/rust-lang/rust/pull/32012 Vecs and VecDeques now drop faster ;-)
https://github.com/rust-lang/rust/pull/32007 compiletest gets some more capabilities
https://github.com/rust-lang/rust/pull/31999 slice.as_ptr generates simpler LLVM IR
https://github.com/rust-lang/rust/pull/31996 libc update
https://github.com/rust-lang/rust/pull/31989 Patterns can now be pointers-to-pointers to str with different lifetimes
https://github.com/rust-lang/rust/pull/31985 libstd can now be compiled with emscripten
https://github.com/rust-lang/rust/pull/31962 One less internal compiler error with trait consts
https://github.com/rust-lang/rust/pull/31945 implement RFC 1461
https://github.com/rust-lang/rust/pull/31928 Stabilized APIs for 1.8
https://github.com/rust-lang/rust/pull/31937 Positional fields are now named starting from "0", "1", ..
https://github.com/rust-lang/rust/pull/31797 vec![..] no longer copies from the stack
https://github.com/rust-lang/rust/pull/31430 zeroing on drop
https://github.com/rust-lang/rust/pull/31335 AsciiExt::in_ascii_{lower, upper}case
https://github.com/rust-lang/rust/pull/30884 Implement inclusive ranges

**Comment by cybergeek94** [comment]
How did preferences make CotW if it was never nominated? Is selection arbitrary now?

**Comment by nasa42** [comment]
We still accept nominations. I'm assuming because of lack of nominations, @llogiq himself nominated this week's CotW.

**Comment by cybergeek94** [comment]
There was a single nomination but it didn't get any votes. There was no bumping comment like previous voting cycles, actually hasn't been for a month or so.

**Comment by nasa42** [comment]
I ask for nomination for next week's CotW is every single issue of TWiR. :-)

**Comment by cybergeek94** [comment]
Ah, ok. Can you make sure [my announcement for `multipart`](https://users.rust-lang.org/t/rfc-mentoring-multipart-0-5-0-more-integrations-better-more-convenient-apis/4881) makes this week's issue? Thanks. :+1:

**Comment by nasa42** [comment]
@cybergeek94 `multipart` has already been mentioned in an old TWiR issue! :-)
if you are looking for contributors and also mentoring as your post mentions, I'd suggest you to take a look at [Call for Participation](https://users.rust-lang.org/t/twir-call-for-participation/4821) thread. Please create a separate issue per task as well as add a east/medium/hard label and I'll included them in the next TWiR issue!

**Comment by nasa42** [comment]
Thanks @llogiq, those PR annotations are awesome!

**Comment by llogiq** [comment]
My pleasure.

@cybergeek94 I was on mobile for some time, so I didn't get around to nag you folks, and since there was only one self-reported crate without votes, I took it upon myself to choose. I'll see if I find the time tomorrow to write a nag message to rust-users and /r/rust.

**Comment by cybergeek94** [comment]
@nasa42 So only one mention, ever, with no statute of limitations? That was back in December 2014 with the first release. What about "Project Updates"?

**Comment by bluss** [comment]
I think using editorial discretion for crates, when needed, is great @llogiq

**Comment by llogiq** [comment]
I think I should go back to nagging. It also gives me more of those internet points to buy cat pictures with or something...

**Comment by cybergeek94** [comment]
@nasa42 I couldn't help but notice that Leaf got an update mention this issue even though it was released back in November, and got a mention then. Several others have been mentioned multiple times as well. If you didn't want to edit this week's issue this late that's all you had to say, I would have been fine waiting for the next one. 

I've had multiple projects, new and updated, get forgotten from TWiR in recent memory, so it's really hard not to take this personally.

**Comment by nasa42** [comment]
@cybergeek94 First of all, I'm sorry that my actions (or lack thereof) made you feel that way.

If you think that I missed mentioning some of your projects, it would be mostly because I missed your project announcements or it didn't meet notability requirements*. It is far more likely that I missed your project announcements on Reddit or on Rust forums. But you can always tweet to us to ensure it gets our attention! (or create a PR).

I usually announce a project only once, and after that only project updates. The [Leaf mention](https://github.com/autumnai/leaf/blob/063ce978004b8bf4b7fc7481fdd58fed0515cfd8/RELEASE.md) is about a project update which happened last week. It's hard to ignore it because of sheer amount of upvotes on its [Reddit thread](https://www.reddit.com/r/rust/comments/48xjyw/announcing_leaf_02_fastest_machine_learning/).

When I looked at [your announcement](https://users.rust-lang.org/t/rfc-mentoring-multipart-0-5-0-more-integrations-better-more-convenient-apis/4881), I noticed "mentoring" word in the title and assumed that it would be a better fit for _Call for Participation_ and suggested you to create separate issues per task so that I can include it in that section. Now, that I read your announcement post again and yes it is something that should be mentioned in _Project Updates_. Can you please make a PR for it to get included to the next issue?

Please also make PRs for your projects which have not yet been included in TWiR. I suggest you to announce one project per week and keep your project announcements distributed across multiple weeks for maximum exposure.

*I do not have specific guidelines of notability yet, but given the (comparatively) small size of Rust community, the requirements are very relaxed. Basically, it should have a proper README, _good enough_ number of stars on GitHub, or _good enough_ votes on announcement post on Reddit, or I've personally found the project interesting, or author has directly asked TWiR for mention.
